tests/testthat/test-carehomes-multistrain.R

```diff
@@ -453,49 +610,162 @@ test_that("Cannot calculate Rt for multistrain without correct inputs", {
+  S <- y[index_S, , ]
+  R <- y[index_R, , ]
+  prob_strain <- y[index_prob_strain, , ]
+
+  rt <- carehomes_Rt(steps, S[, 1, ], p, prob_strain[, 1, ], R = R[, 1, ])
+  expect_equal(dim(rt$eff_Rt_all), c(85, 2))
+  expect_equal(class(rt), c("multi_strain", "Rt"))
+
+  rt_traj <- carehomes_Rt_trajectories(steps, S, p, prob_strain, R = R)
+  expect_equal(dim(rt_traj$eff_Rt_all), c(85, 2, 3))
+  expect_equal(class(rt_traj), c("multi_strain", "Rt_trajectories", "Rt"))
+
+  rt_strain_weighted <-
+    carehomes_Rt(steps, S[, 1, ], p, prob_strain[, 1, ], R = R[, 1, ],
+                 weight_Rt = TRUE)
+  expect_equal(length(rt_strain_weighted$eff_Rt_all), 85)
+  expect_equal(class(rt_strain_weighted), c("single_strain", "Rt"))
+
+  rt_traj_strain_weighted <-
+    carehomes_Rt_trajectories(steps, S, p, prob_strain, R = R,
+                              weight_Rt = TRUE)
+  expect_equal(dim(rt_traj_strain_weighted$eff_Rt_all), c(85, 3))
+  expect_equal(class(rt_traj_strain_weighted),
+               c("single_strain", "Rt_trajectories", "Rt"))
+
+  nms <- names(rt)
+
+  avg_rt <- wtmean_Rt(rt, prob_strain[, 1, ])
+  avg_rt_traj <- wtmean_Rt(rt_traj, prob_strain)
+
+  ## here the 1st strain has weight 1 all along
+  ## (except step 1 --> to investigate)
+  ## so expect the average R to be the same as R for strain 1
+  expect_equal(rt$eff_Rt_general[, 1], avg_rt$eff_Rt_general)
+  expect_equal(rt$eff_Rt_all[, 1], avg_rt$eff_Rt_all)
+  expect_equal(rt$Rt_general[, 1], avg_rt$Rt_general)
+  expect_equal(rt$Rt_all[, 1], avg_rt$Rt_all)
+
+  expect_equal(rt_traj$eff_Rt_general[, 1, ],
+               avg_rt_traj$eff_Rt_general[, ])
+  expect_equal(rt_traj$eff_Rt_all[, 1, ], avg_rt_traj$eff_Rt_all[, ])
+  expect_equal(rt_traj$Rt_general[, 1, ], avg_rt_traj$Rt_general[, ])
+  expect_equal(rt_traj$Rt_all[, 1, ], avg_rt_traj$Rt_all[, ])
+
+  ## the average should be the same if calculated inside the Rt calculation
+  ## functions or post hoc
+  expect_equal(rt_strain_weighted$eff_Rt_general,
+               avg_rt$eff_Rt_general)
+  expect_equal(rt_strain_weighted$eff_Rt_all,
+               avg_rt$eff_Rt_all)
+  expect_equal(rt_strain_weighted$Rt_general,
+               avg_rt$Rt_general)
+  expect_equal(rt_strain_weighted$Rt_all,
+               avg_rt$Rt_all)
+
+  expect_equal(names(avg_rt), nms)
+  expect_equal(names(avg_rt_traj), nms)
+
+  expect_error(wtmean_Rt(1L), "must inherit")
+  expect_error(wtmean_Rt(structure(list(Rt_all = matrix(1)), class = "Rt"),
+                         prob_strain),
+              "Expect elements of Rt to have dimensions")
 })
 
 
 test_that("Can calculate Rt with an empty second variant ", {
   ## Run model with 2 variants, but both have same transmissibility
   ## no seeding for second variant so noone infected with that one
   p <- carehomes_parameters(sircovid_date("2020-02-07"), "england",
-                            strain_transmission = c(1, 1))
+                            strain_transmission = c(1, 0),
```

Comment:
I think this slightly changes the nature of the test - why did you have to set the second element of strain_transmission to zero? should still work with 1?